doc/developer/temporal-filters.md

```diff
@@ -0,0 +1,67 @@
+# Temporal Filters
+
+Materialize is able to introduce and remove records as a function of timestamps presented as data in the records.
+This is performed using idiomatic SQL, with some restrictions, but it can be hard to understand at first.
+
+Consider the following SQL query:
+```sql
+SELECT *
+FROM data
+WHERE mz_logical_timestamp() BETWEEN data.valid_from AND data.valid_until;
+```
+The `WHERE` clause here relates two columns in the data to `mz_logical_timestamp()`, a function that returns the current logical timestamp.
+For one-shot queries `mz_logical_timestamp()` is simply evaluated and replaced by that timestamp when the query executes.
+In a *maintained* query, as for example when you call `CREATE MATERIALIZED VIEW` or `CREATE INDEX`, the value of the function changes every millisecond.
+In particular, in a maintained query any row will not be present at timestamps before its `valid_from` column nor after `valid_until` column.
+
+Materialize does not re-evaluate the predicate every millisecond for every row.
+Understanding what Materialize does do may help you understand the behavior of and restrictions on temporal filters.
+
+## Manipulating timestamped updates
+
+Materialize records *changes* to collections of data.
+Each change occurs at some "logical timestamp", an explicit transcription of the time at which a change should occur.
+Generally, an update occurs at its logical timestamp and is then in effect indefinitely.
+To undo an update one issues a new update with a negated sign, at a future logical timestamp.
```

Comment:
I'm not sure who this doc is for -- if its for mz engineers this is fine but if its for external people too you might want to spell out "an update is a triple of (row, time, diff)"

Reply:
Will do.